jwst/resample/resample.py

```diff
@@ -269,9 +271,32 @@ def resample_many_to_many(self):
 
             if not self.in_memory:
                 # Write out model to disk, then return filename
+                # This sometimes causes problems in operations because
+                # the output file isn't always fully written before the
+                # execution continues
                 output_name = output_model.meta.filename
-                output_model.save(output_name)
-                log.info(f"Saved model in {output_name}")
+                filesavedOK = False
```

Comment:
is it possible to split all of this new logic into its own function in order to enhance readability of `resample_many_to_many()`?

Reply:
I agree with @emolter here. It would be nice to perhaps package up the next 20-some lines that do the actual file saving and associated checking into its own little function that just gets called at this point in the main step code.

Reply:
Latest commit has this broken out into a function